### ISSUE TITLE
When cgeo has no GPS fix, it reverts back to GC.com home coordinates

### ISSUE THREAD

**Issue by dcjkfgdjhd** [cgeo/cgeo#15975]
### Describe your problem!

It happens to me very often, e.g. when hiking in the mountains, when I close cgeo, and later go back to it, that it will not display the map of my current location but instead for a few seconds the location of my GC.com home coordinates.

This is really irritating when wanting to use the map for e.g. checking whether you are on the right path.

I assume that this is caused by cgeo not having a GPS fix, and defaulting to the GC.com home coordinates.

### How to reproduce?

See above.

### Expected result after these steps?

cgeo should always cache and use the **last known** coordinates, when it has not GPS fix.

It should only revert back to the GC.com home coordinates when nothing is cached (e.g. when used for the very first time).

### Reproducible

Unclear

### c:geo Version

2024-07-07

**Comment by moving-bits** [comment]
c:geo uses the last known map coordinates when "follow my location" is disabled. 

When "follow my location" is enabled, it uses the GPS info from Android (which normally makes uses of "position last known to Android" if it does not receive a GPS update currently) and only reverts to "GC home coordinates" if nothing comes up here. Strange that this does not work on your device.

**Comment by dcjkfgdjhd** [comment]
> When "follow my location" is enabled, it uses the GPS info from Android (which normally makes uses of "position last known to Android" if it does not receive a GPS update currently)

Yes, that seems to be the issue. I have a Google Pixel 6a, so one of the most "standard" phones you can have.

Next time I have the problem I will check if this is maybe associated with energy savings mode?

**Comment by moving-bits** [comment]
Did you have the chance to cross-check with energy savings mode switched to off?

**Comment by dcjkfgdjhd** [comment]
Actually I haven't noticed the issue in recent weeks... Will report back if I encounter it again.

**Comment by dcjkfgdjhd** [comment]
I have not observed this problem for a long time now. Not sure what the reason is, but I am going to close this issue.

**Comment by dcjkfgdjhd** [comment]
Minor update: Since my last message I had this happen to me exactly one more time. No problem, obviously, when it happens so rarely, but whatever is causing it is not completely gone.
